### PR TITLE
Enable fracture problems without boundary conditions

### PR DESCRIPTION
**Pull request by streeve**

### CODE REVIEW

File: src/CabanaPD_Solver.hpp [hunk 1]
```diff
@@ -192,15 +192,13 @@ class SolverElastic
...
+        // Only computed once.
         force->computeWeightedVolume( *particles, *neighbors,
                                       neigh_iter_tag{} );
         comm->gatherWeightedVolume();
-        // Compute/communicate LPS dilatation (does nothing for PMB).
-        force->computeDilatation( *particles, *neighbors, neigh_iter_tag{} );
-        comm->gatherDilatation();
-
         // Compute initial forces.
-        computeForce( *force, *particles, *neighbors, neigh_iter_tag{} );
```
Comment (pabloseleson) [inline src/CabanaPD_Solver.hpp:203]: What is the main difference between "computeForce" and "updateForce"?

File: src/CabanaPD_Solver.hpp [hunk 2]
```diff
@@ -255,14 +253,8 @@ class SolverElastic
...
 
-            // Do not need to recompute LPS weighted volume here without damage.
-            // Compute/communicate LPS dilatation (does nothing for PMB).
-            force->computeDilatation( *particles, *neighbors,
-                                      neigh_iter_tag{} );
-            comm->gatherDilatation();
-
-            // Compute internal forces.
-            computeForce( *force, *particles, *neighbors, neigh_iter_tag{} );
+            // Update force and related quantities.
```
Comment (pabloseleson) [inline src/CabanaPD_Solver.hpp:256]: Do we need the same comment "// Update force and related quantities." before Line 200, etc.

File: src/CabanaPD_Solver.hpp [hunk 3]
```diff
@@ -555,16 +551,8 @@ class SolverFracture
             // Update ghost particles.
             comm->gatherDisplacement();
 
-            // Compute/communicate LPS weighted volume (does nothing for PMB).
-            force->computeWeightedVolume( *particles, *neighbors, mu );
```
Comment (pabloseleson) [inline src/CabanaPD_Solver.hpp:559]: Where is computeWeightedVolume now? I do not see it un updateForce.

File: src/CabanaPD_Solver.hpp [hunk 4]
```diff
@@ -580,6 +568,51 @@ class SolverFracture
...
+
+            if constexpr ( std::is_same<typename force_model_type::thermal_type,
+                                        TemperatureDependent>::value )
+                comm->gatherTemperature();
+
+            // Update ghost particles.
+            comm->gatherDisplacement();
+
+            // Update force and related quantities.
+            updateForce();
```
Comment (pabloseleson) [inline src/CabanaPD_Solver.hpp:591]: Bond-breaking is inside the force computation, right? 

Note we need to ensure all bonds break before the force computation, to ensure symmetry of bond-breaking. 

For example, in PDMATLAB2D we have for time integrator:

 % Step 1: Compute velocity at t+dt/2 for all nodes 

% Step 2: Compute displacement at t+dt for all nodes  

**% Update neighbor list if bond breaking occurs**

**% Compute internal force density and macroelastic energy density at t+dt for all nodes**

 % Step 3: Compute velocity at t+dt for all nodes

File: src/CabanaPD_Solver.hpp [hunk 5]
```diff
@@ -580,6 +568,51 @@ class SolverFracture
...
+            integrator->finalHalfStep( *particles );
+
+            output( step );
+        }
+
+        // Final output and timings.
+        this->final_output();
+    }
+
+    void updateForce()
```
Comment (pabloseleson) [inline src/CabanaPD_Solver.hpp:603]: Why do we have "updateForce" here and also in Line 306?

File: src/CabanaPD_Solver.hpp (hunk 1 above)
Comment (streeve) [reply src/CabanaPD_Solver.hpp:203]: Check the new `updateForce` - it calls compute and all other force-related functions

File: src/CabanaPD_Solver.hpp (hunk 2 above)
Comment (streeve) [reply src/CabanaPD_Solver.hpp:256]: It currently says "Compute initial forces" - push a change if you'd like

File: src/CabanaPD_Solver.hpp (hunk 5 above)
Comment (streeve) [reply src/CabanaPD_Solver.hpp:603]: This is a different class

File: src/CabanaPD_Solver.hpp (hunk 3 above)
Comment (streeve) [reply src/CabanaPD_Solver.hpp:559]: This is the next class - see line 603

File: src/CabanaPD_Solver.hpp (hunk 4 above)
Comment (streeve) [reply src/CabanaPD_Solver.hpp:591]: I think this is unrelated - maybe open this as a separate issue

**Review by pabloseleson — APPROVED** [review]

**Pull request merged**